newton/_src/utils/download_assets.py

```
@@ -32,9 +91,27 @@ def _handle_remove_readonly(func, path, exc):
+    lingering file locks.
+    """
+    if not os.path.exists(path):
+        return
+
+    # Try 5 times, with a 1 second delay.
+    # This is to handle Windows [WinError 32] "file in use" errors,
+    # which often happen briefly after a git operation.
+    for i in range(5):
+        try:
+            shutil.rmtree(path, onerror=_handle_remove_readonly)
+            return  # Success
```

Ruff (0.14.3)

Consider moving this statement to an `else` block

(TRY300)